src/database/pipelines/importAiSummaries.ts

```diff
@@ -0,0 +1,92 @@
+import { readFile } from 'fs/promises';
+import { parse } from 'csv-parse';
+import { argv } from 'process';
+import { Kysely, Transaction, sql } from 'kysely';
+import { DB } from '@/database/allDbTypes';
+import { createDB } from '@/database/kyselyDb';
+
+class ImportAiSummaries {
+  public static async run(inputFileName: string, db: Kysely<DB>) {
+    try {
+      await db.transaction().execute(async (trx) => {
+        await new ImportAiSummaries(inputFileName, trx).run();
+      });
+    } catch (error) {
+      console.error(`Error during ImportAiSummaries`, error);
+      throw error;
+    }
+  }
+
+  private constructor(
+    private readonly inputFileName: string,
+    private readonly trx: Transaction<DB>,
+  ) {}
+
+  private async run() {
+    if (!this.inputFileName)
+      throw new Error(`Invalid file name ${this.inputFileName}`);
+
+    const candidateIdLookup = await this.getCandidateIdLookup();
+    const inputStream = await this.createInputStream();
+
+    const summaries = new Array<SummaryRow>();
+    for await (const inputRow of inputStream) {
+      const agendaItemNumber = candidateIdLookup.get(+inputRow.agendaItemId);
+      if (!agendaItemNumber) continue;
+      summaries.push({
+        agendaItemNumber,
+        summary: inputRow.ai_summary,
+      });
+    }
+    console.log('Found usable summaries', summaries);
+    await this.upsertSummaries(summaries);
+    console.log(`Finished, inserted or updated ${summaries.length} summaries`);
+  }
+
+  private async getCandidateIdLookup() {
+    const rows = await this.trx
+      .selectFrom('AgendaItems')
+      .innerJoin('RawAgendaItemConsiderations', (eb) =>
+        eb.onRef('agendaItemNumber', '=', 'reference'),
+      )
+      .groupBy('agendaItemNumber')
+      .having(sql`count(*)`, '=', 1)
```

Comment:
Basically: only summarize cases where the considerations are 1-1 with agenda items

Reply:
Removed this restriction since the considerations may update on their own and break this assumption. So better for the consumer to handle it